core/modules/widgets/widget.js

```diff
@@ -432,28 +439,23 @@ Widget.prototype.assignAttributes = function(domNode,options) {
 				namespace = "http://www.w3.org/1999/xlink";
 				name = name.substr(6);
 			}
-			// Handle styles
-			if(name.substr(0,6) === "style." && name.length > 6) {
-				domNode.style[$tw.utils.unHyphenateCss(name.substr(6))] = value;
-			} else {
-				// Setting certain attributes can cause a DOM error (eg xmlns on the svg element)
-				try {
-					domNode.setAttributeNS(namespace,name,value);
-				} catch(e) {
-				}
+			// Setting certain attributes can cause a DOM error (eg xmlns on the svg element)
+			try {
+				domNode.setAttributeNS(namespace,name,value);
+			} catch(e) {
 			}
 		}
 	}
-	// Not all parse tree nodes have the orderedAttributes property
-	if(this.parseTreeNode.orderedAttributes) {
-		$tw.utils.each(this.parseTreeNode.orderedAttributes,function(attribute,index) {
-			assignAttribute(attribute.name,self.attributes[attribute.name]);
-		});	
-	} else {
-		$tw.utils.each(Object.keys(self.attributes).sort(),function(name) {
-			assignAttribute(name,self.attributes[name]);
-		});	
-	}
+	$tw.utils.each(changedAttributes,function(value,name) {
+		value = self.getAttribute(name);
+		// Check for a prefixed attribute
+		if(name.substr(0,sourcePrefix.length) === sourcePrefix) {
+			domNode.setAttribute(destPrefix + name.substr(sourcePrefix.length),value);
+		// Check for a style attribute
+		} else if(name.substr(0,6) === "style." && name.length > 6) {
```

Comment:
This logic is flawed. If no sourcePrefix is specified, such as in the element widget, then style. attributes do not work.

Reply:
See https://github.com/Jermolene/TiddlyWiki5/pull/7852

Reply:
Thanks @saqimtiaz